File: jar-client/src/main/java/com/beancounter/client/ingest/Filter.java
package com.beancounter.client.ingest;

import com.beancounter.common.model.Asset;
import java.util.ArrayList;
import java.util.Collection;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
public class Filter {

  private final Collection<String> filteredAssets = new ArrayList<>();

  public Filter(@Value("${filter:#{null}}") String filter) {
    init(filter);
  }

  private void init(String filter) {
    if (filter != null) {
      String[] values = filter.split(",");
      for (String value : values) {
        filteredAssets.add(value.toUpperCase());
      }
    }
  }

  public boolean inFilter(Asset asset) {
    if (!filteredAssets.isEmpty()) {
      return filteredAssets.contains(asset.getCode().toUpperCase());
    }
    return true;
  }

  public boolean hasFilter() {
    return !filteredAssets.isEmpty();
  }


}

File: jar-client/src/main/kotlin/com/beancounter/client/ingest/Filter.kt
package com.beancounter.client.ingest

import com.beancounter.common.model.Asset
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import java.util.Locale

/**
 * Determines if the asset code is in a comma separated filter string.
 */
@Service
class Filter(@Value("\${filter:#{null}}") filter: String?) {
    private val filteredAssets: MutableCollection<String> = ArrayList()
    private fun init(filter: String?) {
        if (filter != null) {
            val values = filter.split(",").toTypedArray()
            for (value in values) {
                filteredAssets.add(value.uppercase(Locale.getDefault()))
            }
        }
    }

    /**
     * if a filter has been set, then this will check if the asset code is to be included.
     */
    fun inFilter(asset: Asset): Boolean {
        return if (!filteredAssets.isEmpty()) {
            filteredAssets.contains(asset.code.uppercase(Locale.getDefault()))
        } else true
    }

    /**
     * Hint to determine if we're filtering assets at all.
     */
    fun hasFilter(): Boolean {
        return !filteredAssets.isEmpty()
    }

    init {
        init(filter)
    }
}
